Configure Sentry_SDK

sven-rosenzweig opened the pull request:
Configure Sentry_SDK

The sentry raven library is long deprecated. The python sentry SDK replaces this library.
To get the same behaviour as with raven (only enable the logging with selected loggers via the Handlers), one has to initialise the SDK but manually configure the Logging integration and set the default levels to None.

 Avoiding the sentry configuration in each SAPCC upstream based project (e.g. Neutron or Nova) and furthermore
 in each driver, this has is part of the loci build process. During the build a .pth and .py file will be placed in the
 site_package path. Within this path, an executable line in a .pth file is run at every Python startup.
 In our case, a script configuring sentry if the env variable SENTRY_USE_SDK is
 set. As requirements sentry-sdk must be installed.

sven-rosenzweig committed:
Configure Sentry_SDK

The sentry raven library is long deprecated. The python sentry SDK replaces this library.
To get the same behaviour as with raven
(only enable the logging with selected loggers via the Handlers), one has to initialise
the SDK but manually configure the Logging integration and set the default levels to None.

Avoiding the sentry configuration in each SAPCC upstream based project (e.g. Neutron or Nova) and furthermore
in each driver, this has is part of the loci build process.
During the build a .pth and .py file will be placed in the
site_package path. Within this path, an executable line in a .pth file is run at every Python startup.
In our case, a script configuring sentry if the env variable SENTRY_USE_SDK is
set. As requirements sentry-sdk must be installed.

Co-authored-by: Florian Streibelt  <mail+github@yakshaving.de>

mutax commented:
I got some input from Seba yesterday, I think two things would make sense to add to the `ccsentry.py` script:

1. check `sys.argv[0] `against a whitelist of commands
    -  have a list of strings e.g. neutron, agent, ...
    -  for each string check if its a substring of `argv[0]`
2. add another environment variable to overwrite this list

This would prevent sentry getting enabled with every python process on the target which could otherwise send random logs to sentry, or interfere with debugging maybe. We however need to check if there  are some "hidden" `argv[0]` values if a python subprocess is being created by the agents. 

At the same time we could easily modify the list without building a new image, allowing us to add or remove scripts. That should come in handy when other services like nova want to test that approach.

update: added as extra commit.

mutax committed:
[Setup Sentry] moved script and calling setup from install.sh

Moved the ssentry.py to a file in a subdirectory, so it can be editied
in an editor using syntax highlighting, formatting and autopep

mutax committed:
[Setup Sentry] adding filtering for command names

allow filtering based on argv[0] to enable only for specific commands

mutax committed:
[SENTRY] add some documentation, renaming variable names

changing the prefix of the variables to clearly distinguish
between the namespace sentry is using and our own variables.

mutax commented:
we dropped this approach.
